apps/mosquitto_ctrl/dynsec_client.c

```diff
@@ -56,6 +56,13 @@ int dynsec_client__create(int argc, char *argv[], cJSON *j_command)
 			}
 			password = argv[i+1];
 			i++;
+			request_password = false;
+		} else if (!strcmp(argv[i], "--no-password")) {
+			if (password != NULL) {
+				fprintf(stderr, "Error: -p and --no-password are mutually exclusive.\n");
```

Comment:
You've covered the error case of `-p password --no-password`, but not `--no-password -p password` - could you add that check as well please (checking on `request_password` should do it)

Reply:
Good catch. Will do in a few days.